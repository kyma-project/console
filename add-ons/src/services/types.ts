import { Filters, ConfigurationLabels, Configuration } from '../types';

export interface FiltersService {
  activeFilters: Filters;
  setFilterLabel: (label: string) => void;
  removeFilterLabel: (label: string) => void;
  removeAllFilters: () => void;
  hasActiveLabel: (key: string, value: string) => boolean;
}

export interface LabelsService {
  getUniqLabels: ConfigurationLabels;
}

export interface ConfigurationsService {
  originalConfigs: Configuration[];
  filteredConfigs: Configuration[];
}

export enum SubscriptionType {
  ADD = 'ADD',
  UPDATE = 'UPDATE',
  DELETE = 'DELETE',
}
