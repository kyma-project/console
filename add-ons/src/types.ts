import React from 'react';

export type BackendModule = string;

export interface SecretRef {
  name: string;
  namespace: string;
}

export interface Repository {
  url: string;
  secretRef?: SecretRef;
}

export interface RepositoryAddon {
  name: string;
  status: string;
  version: string;
  reason: string;
  message: string;
}

export interface RepositoryStatus {
  url: string;
  status: string;
  reason: string;
  message: string;
  addons: RepositoryAddon[];
}

export interface ConfigurationStatus {
  phase: string;
  repositories: RepositoryStatus[];
}

export interface Configuration {
  name: string;
  urls: ConfigurationURL[];
  labels: ConfigurationLabels;
  repositories: Repository[];
  status: ConfigurationStatus;
}

export interface ConfigurationLabels {
  [key: string]: string;
}
export type ConfigurationURL = string;

export interface FilterLabels {
  [key: string]: string[];
}

export interface Filters {
  search: string;
  labels: FilterLabels;
}

export interface ActiveFiltersAction {
  payload: {
    key: string;
    value: string;
  };
  type: ActiveFiltersActionType;
}

export enum ActiveFiltersActionType {
  SET_SEARCH = 'SET_SEARCH',
  SET_LABEL = 'SET_LABEL',
  REMOVE_LABEL = 'REMOVE_LABEL',
  REMOVE_ALL_FILTERS = 'REMOVE_ALL_FILTERS',
}

export interface RemoveFiltersInterface {
  removeFilterLabel: (key: string, value: string) => void;
  removeAllFiltersLabels: () => void;
}

export interface FiltersLabelsInterface {
  uniqueLabels: FilterLabels;
  setFilterLabel: (key: string, value: string) => void;
  hasActiveLabel: (key: string, value: string) => boolean;
}

export interface ConfigurationsService
  extends FiltersLabelsInterface,
    RemoveFiltersInterface {}

export interface Notification {
  title: string;
  content: string | React.ReactNode;
  color?: string;
  icon?: string;
}
