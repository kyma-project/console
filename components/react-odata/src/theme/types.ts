export interface ThemeInterface {
  color?: string;
}
