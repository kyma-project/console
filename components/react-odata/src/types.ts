export interface Child {
  attributes: { [key: string]: string };
  name: string;
  value: string;
  children: Child[];
}
