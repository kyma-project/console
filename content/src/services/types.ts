import { ClusterAssetGroup } from '@kyma-project/common';

export interface ClusterAssetGroups {
  [group: string]: ClusterAssetGroup[];
}

export interface Navigation {
  [group: string]: NavigationItem[];
}

export interface NavigationItem {
  name: string;
  group: string;
  displayName: string;
}
