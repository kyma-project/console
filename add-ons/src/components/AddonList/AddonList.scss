.addon-list__error-wrapper {
  padding: 16px;
  font-size: 22px;
  font-weight: 500;
  text-align: center;
}
