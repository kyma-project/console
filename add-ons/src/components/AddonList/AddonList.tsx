import React from 'react';
import { Spinner } from '@kyma-project/components';
import './AddonList.scss';

import AddonPanel from './../AddonPanel/AddonPanel';

import { QueriesService, ConfigurationsService } from '../../services';

import { ERRORS } from '../../constants';

const AddonList: React.FunctionComponent = () => {
  const { error, loading = true } = React.useContext(QueriesService);
  const { configurationsExist, filteredConfigs } = React.useContext(
    ConfigurationsService,
  );

  if (loading) {
    return <Spinner />;
  }
  if (!configurationsExist()) {
    return (
      <div className="addon-list__error-wrapper">
        {ERRORS.RESOURCES_NOT_FOUND}
      </div>
    );
  }
  if (error) {
    return <div className="addon-list__error-wrapper">{ERRORS.SERVER}</div>;
  }
  return (
    <section className="fd-has-margin-m" id="addon-list">
      {filteredConfigs.map(config => (
        <AddonPanel key={config.name} config={config} />
      ))}
    </section>
  );
};

export default AddonList;
