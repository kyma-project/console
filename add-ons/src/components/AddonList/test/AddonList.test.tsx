import React from 'react';
import AddonList from './../AddonList';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Configuration } from '../../../types';

const configMock: Configuration = {
  name: 'configuration',
  urls: [],
  labels: {},
  repositories: [],
  status: {
    phase: 'ready',
    repositories: [],
  },
};

describe('AddonList', () => {
  // mock QueriesService and ConfigurationService
  let useContextMock: any;
  let originalUseContext: any;

  beforeEach(() => {
    useContextMock = React.useContext = jest.fn();
    originalUseContext = React.useContext;
  });

  afterEach(() => {
    React.useContext = originalUseContext;
  });

  it('Renders loading state', () => {
    useContextMock
      .mockReturnValueOnce({ error: null, loading: true })
      .mockReturnValueOnce({ configurationsExist: null, filteredConfigs: [] });

    const component = shallow(<AddonList />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Renders "resource not found" when configuration does not exist after load', () => {
    useContextMock
      .mockReturnValueOnce({ error: null, loading: false })
      .mockReturnValueOnce({
        configurationsExist: () => false,
        filteredConfigs: [],
      });

    const component = shallow(<AddonList />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Renders error there is a query error', () => {
    useContextMock
      .mockReturnValueOnce({ error: 'error!', loading: false })
      .mockReturnValueOnce({
        configurationsExist: () => true,
        filteredConfigs: [],
      });

    const component = shallow(<AddonList />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Renders empty list when there are no configs', () => {
    useContextMock
      .mockReturnValueOnce({ error: null, loading: false })
      .mockReturnValueOnce({
        configurationsExist: () => true,
        filteredConfigs: [],
      });

    const component = shallow(<AddonList />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Renders addon list', () => {
    useContextMock
      .mockReturnValueOnce({ error: null, loading: false })
      .mockReturnValueOnce({
        configurationsExist: () => true,
        filteredConfigs: [configMock],
      });

    const component = shallow(<AddonList />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
