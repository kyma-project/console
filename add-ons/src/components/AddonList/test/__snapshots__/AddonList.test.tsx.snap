// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddonList Renders "resource not found" when configuration does not exist after load 1`] = `
<div
  className="addon-list__error-wrapper"
>
  Resources not found.
</div>
`;

exports[`AddonList Renders addon list 1`] = `
<section
  className="fd-has-margin-m"
  id="addon-list"
>
  <AddonPanel
    config={
      Object {
        "labels": Object {},
        "name": "configuration",
        "repositories": Array [],
        "status": Object {
          "phase": "ready",
          "repositories": Array [],
        },
        "urls": Array [],
      }
    }
    key="configuration"
  />
</section>
`;

exports[`AddonList Renders empty list when there are no configs 1`] = `
<section
  className="fd-has-margin-m"
  id="addon-list"
/>
`;

exports[`AddonList Renders error there is a query error 1`] = `
<div
  className="addon-list__error-wrapper"
>
  Server error. Please contact the admin of the cluster.
</div>
`;

exports[`AddonList Renders loading state 1`] = `<Spinner />`;
